openslides_backend/action/actions/meeting_user/mixin.py

```diff
@@ -1,93 +1,203 @@
-from ....shared.patterns import (
-    FullQualifiedId,
-    fqid_from_collection_and_id,
-    id_from_fqid,
+from openslides_backend.permissions.management_levels import (
+    CommitteeManagementLevel,
+    OrganizationManagementLevel,
 )
+from openslides_backend.permissions.permissions import Permissions
+
+from ....action.action import Action
+from ....shared.exceptions import ActionException, MissingPermission, PermissionDenied
+from ....shared.patterns import fqid_from_collection_and_id
 
 
 class MeetingUserMixin(Action):
+    standard_fields = [
+        "comment",
+        "number",
+        "structure_level",
+        "vote_weight",
+        "personal_note_ids",
+        "speaker_ids",
+        "supported_motion_ids",
+        "submitted_motion_ids",
+        "assignment_candidate_ids",
+        "vote_delegated_vote_ids",
+        "vote_delegated_to_id",
+        "vote_delegations_from_ids",
+        "chat_message_ids",
+    ]
+
+    def check_permissions(self, instance: Dict[str, Any]) -> None:
+        """standard_fields have to be checked for user.can_manage, which is always sufficient and
+        even needed, if there is no data at all exempt the required fields.
+        Special fields like about_me and group_ids could be managed also with other permissions.
+        Details see https://github.com/OpenSlides/OpenSlides/wiki/meeting_user.create"""
+        if any(fname in self.standard_fields for fname in instance.keys()) or not any(
+            fname in ["about_me", "group_ids"] for fname in instance
+        ):
+            return super().check_permissions(instance)
+
+        def get_user_and_meeting_id() -> Tuple[int, int]:
+            fields = ["user_id", "meeting_id"]
+            if any(fname not in instance for fname in fields):
+                mu = self.datastore.get(
+                    fqid_from_collection_and_id("meeting_user", instance["id"]),
+                    ["user_id", "meeting_id"],
+                    lock_result=False,
+                )
+                return cast(Tuple[int, int], ([mu[fname] for fname in fields]))
+            return cast(Tuple[int, int], (instance[fname] for fname in fields))
+
+        def get_request_user_data() -> Dict[str, Any]:
+            return self.datastore.get(
+                fqid_from_collection_and_id("user", self.user_id),
+                ["organization_management_level", "committee_management_ids"],
+                lock_result=False,
+            )
+
+        def get_committee_id() -> int:
+            return self.datastore.get(
+                fqid_from_collection_and_id("meeting", meeting_id),
+                ["committee_id"],
+                lock_result=False,
+            )["committee_id"]
+
+        def raise_own_exception() -> bool:
+            try:
+                super(MeetingUserMixin, self).check_permissions(instance)
+                return False
+            except Exception:
+                return True
```

Comment:
I don't like catching each exception here. Specify, which exception you are expecting.

Reply:
The caught exception doesn't matter, because  the specific exception will be used anyway. This is just a convenience function